spec/facility_spec.rb

```diff
@@ -23,4 +26,150 @@
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
     end
   end
+
+  describe 'Vehicle Registration' do
+    facility_1 = Facility.new({name: 'Albany DMV Office', address: '2242 Santiam Hwy SE Albany OR 97321', phone: '541-967-2014' })
+    facility_2 = Facility.new({name: 'Ashland DMV Office', address: '600 Tolman Creek Rd Ashland OR 97520', phone: '541-776-6092' })
+    cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+    bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev} )
+    camaro = Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice} )
+    
+    it 'attributes should be empty' do
+      expect(facility_1.add_service('Vehicle Registration')).to eq(["Vehicle Registration"])
+      expect(cruz.registration_date).to eq nil
+      expect(facility_1.registered_vehicles).to eq([])
+      expect(facility_1.collected_fees).to eq(0)
+    end
+
+    it '#register_vehicle' do
+      expect(facility_1.registered_vehicles).to eq([])
+
+      facility_1.register_vehicle(cruz)
+
+      expect(facility_1.registered_vehicles).to eq([cruz])
+    end
+
+    it 'registers a vehicle and sets the registration date' do
+      facility_1 = Facility.new({ name: 'Albany DMV Office', address: '2242 Santiam Hwy SE Albany OR 97321', phone: '541-967-2014' })
+      cruz = Vehicle.new({ vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice })
+
+      facility_1.add_service('Vehicle Registration')
+      facility_1.register_vehicle(cruz)
+
+      cruz.get_registered
+
+      expect(cruz.registration_date).to eq(Date.today)
```

Comment:
I see now that you covered this functionality but that you have to call it on the instance of the Vehicle. That makes sense. It would be totally fine to have this registration date update in your facility method but I think I understand where you were coming from (creating methods to manipulate its own attributes maybe). It is sometimes okay to have classes manipulate other object's data when it is a *side effect* of what their method is doing. For example, updating the registration date of a vehicle is a *side effect* of registering a car at a facility.Happy to chat more about this if you'd like